### PR TITLE
Guest customer 1 er

### PR DESCRIPTION
**Pull request by erose357**
Alright guys, bear with me on this one. There's a log in this PR, I went outside of my user story to stay on track for our overall goal. I'm hoping this will prevent extra work later.  I will make sure my future PRs are much smaller.

-adds migrations for the vendors table, including a slug column
-a service for the slug is built so keep in mind we need to run that whenever adding rows to the database
-changes routing so the link from an individual vendor directs to a vendor/items page
-my test is weak, but I'll clean it up in this next user story, I wanted to get this code up so I'm not holding you guys up
-make sure you run the migrations once this is approved and you pull it down

### CODE REVIEW

File: db/migrate/20171018132438_add_image_to_vendors.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddImageToVendors < ActiveRecord::Migration[5.1]
+  def change
+    add_column :vendors, :image, :string
```
Comment (JoelLindow) [inline db/migrate/20171018132438_add_image_to_vendors.rb]: Yes! Needed this.

**Comment by JoelLindow** [comment]
Looks Good. Waiting for 1 more team member to review. Can not move forward on my code until this is merged

File: db/migrate/20171018132438_add_image_to_vendors.rb (hunk 1 above)
Comment (boveus) [reply db/migrate/20171018132438_add_image_to_vendors.rb]: Nice, now we can revive Uncle Frank.

File: app/models/vendor.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Vendor < ApplicationRecord
+  validates_presence_of :slug
```
Comment (boveus) [inline app/models/vendor.rb]: Do we want to validate uniqueness of the slug?

**Comment by erose357** [comment]
I apologize for my oversight earlier.  In the name of progress, I skipped the tests that broke after these changes.  I've created a pivotal tracker card and assigned it to myself to address this.

File: app/models/vendor.rb (hunk 2 above)
Comment (erose357) [reply app/models/vendor.rb]: I left it out so we could discuss.  Thank you for the reminder.  It seems to me that we have to, but I wanted to run it by the group.

File: db/migrate/20171018132438_add_image_to_vendors.rb (hunk 1 above)
Comment (erose357) [reply db/migrate/20171018132438_add_image_to_vendors.rb]: I think we are obligated to revive Uncle Frank.

**Comment by boveus** [comment]
Overall it looks great.  I don't feel like this was too much code to adequately review.  This is a good size for PRs from my perspective.

My only concern is that we may want to validate the uniqueness of slugs as well to avoid any sort of collisions with our route names, but that is kind of a minor concern.

**Pull request merged**